### ISSUE TITLE
Feature request: Refactor orderBy to support multiple fields and relations.

### ISSUE THREAD

**Issue by DregondRahl** [maoosi/prisma-appsync#12]
Was trying multiple orderBy and came into an issue that would throw an error "e[r].toLowerCase is not a function" when I looked into the code, I can see 'input' is being passed but it wouldn't be the object key but the parent array key.

```
query MyQuery {
  listUsers(skip: 2, take: 2, orderBy: {id: DESC, name: DESC}) {
    name
    id
  }
}
```

Prisma supports multiple OrderBy so I made some changes and built it with below: 
```
    private parseOrderBy(orderByInput: any) {
        const orderByOutput: any = []
        for (const input in orderByInput) {
            const orderByKey = Object.keys(orderByInput[input])[0]
            const orderByArg = orderByInput[input][orderByKey].toLowerCase()

            if (PrismaOrderByArgs.includes(orderByArg)) {
                orderByOutput.push({
                    [orderByKey]: orderByArg,
                })
            }
        }

        return orderByOutput
    }
```

Does this approach make sense or is there a different plan for orderBy ?

**Comment by maoosi** [comment]
Ideally, I would like to support both multiple orderBy and orderBy on relations. I’m not yet settled on the best approach.

What you are suggesting could be one solution. A second one could be turning orderBy input into an array, similar to: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#sort-user-by-multiple-fields---email-and-role

A third solution (which sounds better to me, even though not exactly following Prisma syntax) - would be using a similar structure to filters, where you can use the `AND: []` syntax for multiple orderBy and directly target a field or a relation:

```
input PostOrderByInput {
    AND:[PostOrderByInput]
    title: OrderByArg
    author: UserOrderByInput
}
```

> FYI - Early next week, I'll release a new version that introduces pagination and a new where filtering mechanism closer to Prisma (as you suggested in another issue). You can try it via `yarn add prisma-appsync@next`. Quick overview at: https://github.com/maoosi/prisma-appsync/blob/next/CHANGELOG.md#version-100-beta55

**Comment by DregondRahl** [comment]
With the code above, it accepts the orderBy in this way 
`orderBy: {id: DESC, name: DESC}`

But converts it to prisma syntax 
`orderBy: [{ id: DESC }, { name: DESC }];`

Which for me looks good because I won't need to nest the orderBy items, however if we can support more similar syntax as prisma and nexus (second option) it will be easier for developers to shift from nexus if they need to deploy with AppSync.

The third option, looks good, but I'm not familiar with that syntax being used anywhere else. Though it would be clear the order at which records will be sorted in. 



Regarding the pagination update, I'm just working on a few things so it was easy enough to add while I checked on a few integrations. This package is honestly a life saver! 

I added count() also to help with offset pagination

```
    public async count(model: string, args: RequestProps) {
        await this.runBeforeResolveHook({
            operation: Operations.count,
            model,
            args,
        })

        if (process.env.JEST_WORKER_ID) return args

        // count all records matching the where arguments
        const results = await this.prisma[model].count({
            ...(args.where && { where: args.where }),
            ...(args.skip && { skip: args.skip }),
            ...(args.take && { take: args.take }),
            ...(args.cursor && { cursor: args.cursor }),
            ...(args.distinct && { distinct: args.distinct }),
            ...(args.orderBy && { orderBy: args.orderBy }),
        })

        await this.runAfterResolveHook({ result: results })

        return results
    }
```

**Comment by maoosi** [comment]
Thanks for sharing your thoughts on this! I’ll add both refactoring of `orderBy` and adding a `count` query to the roadmap.

Before diving into new features, I will spend some time refactoring the Tests and adding some CI/CD automation. After this is in place, it will make it easier to directly submit pull requests, in case you would like to contribute.

**Comment by maoosi** [comment]
Both implemented in [1.0.0-beta.58.2](https://github.com/maoosi/prisma-appsync/releases/tag/1.0.0-beta.58.2) (see `orderBy` query filter and `count` query).
